DOCSP-37861: gridfs tutorial (#17)
mongodb/docs-java-rs: source/tutorials/gridfs.txt

.. important::

   This guide uses the ``Subscriber`` implementations, which are
   described in the :ref:`Quick Start Primer <javars-primer>`.

Connect to a MongoDB Deployment
-------------------------------

First, connect to a MongoDB deployment and declare and define
a ``MongoDatabase`` instance.

The following code connects to a standalone
MongoDB deployment running on ``localhost`` on port ``27017``:

.. code-block:: java

   MongoClient mongoClient = MongoClients.create();

To learn more about connecting to MongoDB deployments,
see the :ref:`javars-connect` tutorial.

Create a GridFS Bucket
----------------------

GridFS stores files in two collections:

- ``chunks``: stores the file chunks
- ``files``: stores file metadata

The two collections are in a common bucket and the collection names
are prefixed with the bucket name.

The driver provides the ``GridFSBuckets.create()`` method to
create ``GridFSBucket`` instances:

.. code-block:: java

   MongoDatabase myDatabase = mongoClient.getDatabase("mydb");
    
   // Create a gridFSBucket using the default bucket name "fs"
   GridFSBucket gridFSBucket = GridFSBuckets.create(myDatabase);

> You can specify a bucket name to ``GridFSBuckets.create()`` method:

Reviewer: ```suggestion
You can pass a bucket name to ``GridFSBuckets.create()`` method:
```

.. code-block:: java

   // Create a gridFSBucket with a custom bucket name "files"
   GridFSBucket gridFSFilesBucket = GridFSBuckets.create(myDatabase, "files");

.. note::

   GridFS automatically creates indexes on the ``files`` and ``chunks``

>    collections upon upload of data into the GridFS bucket.

Reviewer: ```suggestion
   collections when you upload data to the GridFS bucket.
```
